securedrop/passphrases.py

```diff
@@ -0,0 +1,123 @@
+        self._random_generator = SystemRandom()
+
+        self._fallback_language = fallback_language
+        self._language_to_words = language_to_words
+        if self._fallback_language not in self._language_to_words:
+            raise InvalidWordsListError(
+                "Missing words list for fallback language {}".format(self._fallback_language)
+            )
+
+        # Validate each words list
+        for language, words_list in self._language_to_words.items():
+            # Ensure all words are ascii
+            try:
+                " ".join(words_list).encode("ascii")
+            except UnicodeEncodeError:
+                raise InvalidWordsListError("A words list contains non-ASCII words.")
+
+            # Ensure that passphrases longer than what's supported can't be generated
+            longest_word = max(words_list, key=len)
+            longest_passphrase_length = len(longest_word) * self._PASSPHRASE_WORDS_COUNT
+            longest_passphrase_length += self._PASSPHRASE_WORDS_COUNT  # One space between each word
+            if longest_passphrase_length >= self._MAX_PASSPHRASE_LENGTH:
+                raise InvalidWordsListError(
+                    "Passphrases over the maximum length ({}) may be generated:"
+                    " longest word in words list for language {} is '{}' and number of words per"
+                    " passphrase is {}".format(
+                        self._MAX_PASSPHRASE_LENGTH,
+                        language,
+                        longest_word,
+                        self._PASSPHRASE_WORDS_COUNT,
+                    )
+                )
+
+            # Ensure that passphrases shorter than what's supported can't be generated
+            shortest_word = min(words_list, key=len)
+            shortest_passphrase_length = len(shortest_word) * self._PASSPHRASE_WORDS_COUNT
+            shortest_passphrase_length += self._PASSPHRASE_WORDS_COUNT
+            if shortest_passphrase_length <= self._MIN_PASSPHRASE_LENGTH:
+                raise InvalidWordsListError(
+                    "Passphrases under the minimum length ({}) may be generated:"
+                    " shortest word in words list for language {} is '{}' and number of words per"
+                    " passphrase is {}".format(
+                        self._MIN_PASSPHRASE_LENGTH,
+                        language,
+                        shortest_word,
+                        self._PASSPHRASE_WORDS_COUNT,
+                    )
+                )
+
+    @classmethod
+    def get_default(cls) -> "PassphrasesGenerator":
+        global _current_generator
+        if _current_generator is None:
+            language_to_words = _parse_available_words_list(Path(config.SECUREDROP_ROOT))
+            # TODO: The french words list is problematic because it has really short words
+            # TODO: Remove this after fixing the words list
+            del language_to_words["fr"]
```

Comment:
If `generate_passphrase` is pickier about what it produces, we don't have to lose French. :slightly_smiling_face: